docs/rule/no-mut-helper.md

```diff
@@ -0,0 +1,244 @@
+# no-mut-helper
+
+## Reasons to not use [the `mut` helper](https://api.emberjs.com/ember/release/classes/Ember.Templates.helpers/methods/each?anchor=mut)
+
+1. The mut helper is non-intuitive to use, see, teach, and learn since it can either be a getter or a setter based on the context in which it’s used.
+
+Example:
+
+```hbs
+{{#let (mut this.foo) as |foo|}}
+  <!-- When used like this, it's a getter -->
+  {{foo}}
+
+  <!-- When used like this, it's a setter -->
+  <button {{action foo 123}}>Update Foo</button>
+{{/let}}
+```
+
+The need for the [no-extra-mut-helper-argument](no-extra-mut-helper-argument.md) rule is further evidence that `mut` has a non-intuitive signature and frequently gets misused.\
+2. The mut helper is usually only used as a pure setter, in which case there are other template helpers that are pure setters that could be used instead of mut (e.g. [ember-set-helper](https://github.com/pzuraq/ember-set-helper)).\
+3. The mut helper can re-introduce 2 way data binding into Glimmer Components on named arguments where a child can change a parent’s data, which goes against the Data Down Actions Up principle, goes against Glimmer Components’ intention to have immutable arguments, and is [discouraged by the Ember Core team](https://www.pzuraq.com/on-mut-and-2-way-binding/).
+
+Example:
+
+```hbs
+<input
+  type="checkbox"
+  checked={{@checked}}
+  {{on "change" (fn (mut @checked) (not @checked))}}
+/>
+```
+
+## What this rule does
+
+This rule forbids any use of the `mut` helper, both as a getter and a setter, in any context. It also
+surfaces possible alternatives in the lint violation message to help guide engineers to resolving
+the lint violations.
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+<MyComponent @isDropdownOpen={{mut this.isDropdownOpen}}/>
+```
+
+```hbs
+<button {{action (mut this.isDropdownOpen) false}}>Close Dropdown</button>
+```
+
+```hbs
+<MyComponent @closeDropdown={{action (mut this.isDropdownOpen) false}}/>
+```
+
+```hbs
+<MyComponent onclick={{fn (mut this.isDropdownOpen) false}}/>
+```
+
+```hbs
+<Input onchange={{action (mut this.profile.description) value="target.value"}}/>
+```
+
+```hbs
+{{my-component click=(action (mut this.isDropdownOpen) false)}}
+```
+
+```hbs
+{{my-component value=(mut this.profile.description)}}
+```
+
+This rule **allows** the following:
+
+```hbs
+<MyComponent @isDropdownOpen={{this.isDropdownOpen}}/>
+```
+
+```hbs
+<button {{action (ember-set-helper$set this "isDropdownOpen" false)}}>Close Dropdown</button>
```

Comment:
We should probably show the standard (not Batman/Wall Street) syntax here, since very few apps and addons are using that syntax.

Reply:
thanks for reviewing, Chris. ya, good point, agreed. fixed throughout the Pull Request, thanks for the suggestion.